presets/presetManager.py

```diff
@@ -0,0 +1,223 @@
+import os.path
+import logging
+import json
+
+logger = logging.getLogger(__name__)
+
+
+class PresetManager(object):
+    MAX_DEPTH = 2
+
+    def __init__(self, paths, strict=False):
+        self.presets = {}
+        self.strict = strict
+
+        # normalize paramter type string -> array
+        if isinstance(paths, basestring):
+            paths = [paths]
+
+        self._loadPaths(paths)
+
+    def _loadPaths(self, paths, depth=0):
+        if depth > self.MAX_DEPTH:
+            return
+
+        for i in range(len(paths)):
+            path = paths[i]
```

Comment:
serious about what? :)

Iterating on a list is 99.9999% of times done using the iteration builtin. Recurring to other method is commonly done when you have the need to _change_ the object you are iterating upon, or other not-very-clean usage of iterations.

Since there is a very procedural (in contrast to functional) style in the code, it's been hard for me to follow the code. I expected that inside some other method you were changing the paths, or stuff like that.

Now I guess this is not the case, so please write idiomatically!